flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/active/ActiveResourceManager.java

```diff
@@ -235,6 +241,8 @@ public void onPreviousAttemptWorkersRecovered(Collection<WorkerType> recoveredWo
         for (WorkerType worker : recoveredWorkers) {
             final ResourceID resourceId = worker.getResourceID();
             workerNodeMap.put(resourceId, worker);
+            previousAttemptUnregisteredWorkers.add(resourceId);
```

Comment:
That's a good point.

I don't think a standby RM will re-request workers, because it only re-request workers when there are unfulfilled resource requirements, which a standby RM should not have since it doesn't accept any resource requirement until becoming the leader. However, it is indeed possible that the standby RM tries to stop the worker. I'll add a check to make sure it only tries to stop workers when it has the leadership.